projects/georeview/index.js

```diff
@@ -0,0 +1,250 @@
+const ymaps = window.ymaps;
+const swiper = window.Swiper;
+
+const hasOwnProperty = Object.prototype.hasOwnProperty;
+
+ymaps.ready(init);
+
+const storageName = 'geo-review';
+
+function init() {
+  const map = new ymaps.Map('map', {
+    center: [54.19, 37.61],
+    zoom: 13,
+  });
+
+  // создаём кластер для поддержки кластеризации
+  const clusterer = new ymaps.Clusterer({
+    // groupByCoordinates: true,
+    clusterDisableClickZoom: true,
+    clusterOpenBalloonOnClick: false,
+  });
+  // добавляем обработчик кликов по кластеру
+  clusterer.events.add('click', (e) => {
+    const target = e.get('target');
+    const coords = target.geometry.getCoordinates();
+
+    openReviews(coords, target);
+  });
+  // добавляем обработчик кликов по карте
+  map.events.add('click', (e) => {
+    const coords = e.get('coords');
+
+    openBalloon(coords, createForm(coords));
+  });
+
+  // добавляем наш кластер на карту
+  map.geoObjects.add(clusterer);
+
+  // открыть балун
+  const openBalloon = async (coords, content) => {
+    map.balloon.open(coords, content);
+    document.title = await getAddress(coords);
+  };
+
+  // закрыть балун
+  const closeBalloon = () => {
+    map.balloon.close();
+  };
+
+  // открыть отзывы
+  const openReviews = (coords, targetClusterer) => {
+    // если нажатый кластер имеет несколько кластеров
+    if (hasOwnProperty.call(targetClusterer, '_clusterListeners')) {
+      // начинаем делать грязЪ
+      const places = {};
+      // перебираем кластеры
+      targetClusterer.getGeoObjects().forEach((obj) => {
+        // формируем данные под карусельку
+        const objectCoordiates = obj.geometry.getCoordinates();
+        const coordinateId = JSON.stringify(objectCoordiates);
+        const coordsData = storageGetData(objectCoordiates);
+        const placeData = {
+          data: coordsData,
+          address: coordsData[0].address,
+          coords: objectCoordiates,
+          coordsId: coordinateId,
+        };
+        places[objectCoordiates] = placeData;
```

Comment:
И здесь можно не создавать перменную:
```suggestion
        places[objectCoordiates] = {
          data: coordsData,
          address: coordsData[0].address,
          coords: objectCoordiates,
          coordsId: coordinateId,
        };
```